flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -326,6 +346,52 @@ class CodeGenerator(
   }
 
   /**
+    * Generates a [[TableFunctionCollector]] that can be passed to Java compiler.
+    *
+    * @param name Class name of the table function collector. Must not be unique but has to be a
+    *             valid Java class identifier.
+    * @param bodyCode body code for the collector method
+    * @param returnType The type information of the element collected by the collector
+    * @return instance of GeneratedFunction
+    */
+  def generateTableFunctionCollector(
+    name: String,
+    bodyCode: String,
+    returnType: TypeInformation[Any])
+  : GeneratedFunction[TableFunctionCollector[Any]] = {
```

Comment:
I would introduce a new `GeneratedCollector` class as this is actually not a Flink `Function`.